internal/core/data/application/reading.go

```diff
@@ -216,3 +220,48 @@ func ReadingsByDeviceNameAndResourceNamesAndTimeRange(deviceName string, resourc
 	}
 	return readings, totalCount, nil
 }
+
+// AsyncPurgeReading purge readings and related events according to the retention capability. This function is implemented to be run as a separate gorountine schedule job.
+func AsyncPurgeReading(interval time.Duration, ctx context.Context, dic *di.Container) {
```

Comment:
since the naming is async, you should write the go routine logic in this function.
also, wrap this function with sync.Once to make sure it can only be trigger once.

Reply:
Refactored and added sync.Once.